From 213d5dea434c3d4d80552a79d3894ecea12215d7 Mon Sep 17 00:00:00 2001
From: Qiang Yu <Qiang.Yu@amd.com>
Date: Thu, 10 Aug 2017 15:57:09 +0800
Subject: [PATCH 44/46] amdgpu: [HYBRID] change to use amdgpu_bo_free

amdgpu_bo_reference is dropped in upstream, some hybrid
spec code still use it. So change it to amdgpu_bo_free.

This patch should be splited and merged to the coresponding
hybrid patches which introduce the usage of amdgpu_bo_reference
when next rebase.

Change-Id: I79ca028270adfc8427cf3d3e6e18002c6c7590b4
Signed-off-by: Qiang Yu <Qiang.Yu@amd.com>
Reviewed-by: Junwei Zhang <Jerry.Zhang@amd.com>
---
 amdgpu/amdgpu_bo.c    | 4 ++--
 amdgpu/amdgpu_vamgr.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/amdgpu/amdgpu_bo.c b/amdgpu/amdgpu_bo.c
index 09028c7..1d90e5e 100644
--- a/amdgpu/amdgpu_bo.c
+++ b/amdgpu/amdgpu_bo.c
@@ -1004,7 +1004,7 @@ int amdgpu_bo_va_op_refcounted(amdgpu_device_handle dev,
 				}
 
 				/* Just drop the reference. */
-				amdgpu_bo_reference(&vao->bo, NULL);
+				amdgpu_bo_free(&vao->bo);
 				/* remove the remap from list */
 				list_del(&vao->list);
 				free(vao);
@@ -1043,7 +1043,7 @@ int amdgpu_bo_va_op_refcounted(amdgpu_device_handle dev,
 		/* unref the bo right after the unmap call */
 		if (ops == AMDGPU_VA_OP_UNMAP) {
 			/* Just drop the reference. */
-			amdgpu_bo_reference(&bo, NULL);
+			amdgpu_bo_free(&bo);
 			/* remove the remap from list */
 			list_del(&vao->list);
 			free(vao);
diff --git a/amdgpu/amdgpu_vamgr.c b/amdgpu/amdgpu_vamgr.c
index 5c91d58..6e0d7e3 100644
--- a/amdgpu/amdgpu_vamgr.c
+++ b/amdgpu/amdgpu_vamgr.c
@@ -499,7 +499,7 @@ int amdgpu_va_range_free(amdgpu_va_handle va_range_handle)
 			if (r)
 				return -EINVAL;
 			/* Just drop the reference. */
-			amdgpu_bo_reference(&vahandle->bo, NULL);
+			amdgpu_bo_free(&vahandle->bo);
 			/* remove the remap from list */
 			list_del(&vahandle->list);
 			free(vahandle);
-- 
2.7.4

